code/__HELPERS/path.dm

```diff
@@ -0,0 +1,379 @@
+			path.Swap(i,length(path)-i+1)
+	openc = null //cleaning after us
+	sources = null
+	caller.calculating_path = FALSE
+	return path
+
+/// Called when we've hit the goal with the node that represents the last tile, then sets the path var to that path so it can be returned by [datum/pathfind/proc/start_search]
+/datum/pathfind/proc/unwind_path(datum/jps_node/unwind_node)
+	success = unwind_node.tile == end
+	path = new()
+	var/turf/iter_turf = unwind_node.tile
+	path.Add(iter_turf)
+
+	var/list/b = list(iter_turf)
+	var/turf/goal_turf = sources[iter_turf]
+	while(goal_turf && goal_turf != PATH_START)
+		b.Add(goal_turf)
+		while(iter_turf != goal_turf)
+			var/turf/next_turf = get_step_towards(iter_turf, goal_turf)
+			iter_turf = next_turf
+			path.Add(iter_turf)
+		goal_turf = sources[iter_turf]
+
+/// For performing a scan in a given lateral direction
+/datum/pathfind/proc/lateral_scan_spec(turf/original_turf, heading)
+	var/steps_taken = 0
+	var/datum/jps_node/unwind_node = openc[original_turf]
+	while(!unwind_node)
+		var/turf/older = sources[original_turf]
+		if(!older)
+			CRASH("JPS error: Lateral scan couldn't find a home node")
+		unwind_node = openc[older]
+
+	var/turf/current_turf = original_turf
+	var/turf/lag_turf = original_turf
+
+	while(TRUE)
+		if(path) // lazy way to force out when done, do better
+			return
+		lag_turf = current_turf
+		current_turf = get_step(current_turf, heading)
+		steps_taken++
+		if(!CAN_STEP(lag_turf, current_turf))
+			return
+
+		var/closeenough
+		if(mintargetdist)
+			closeenough = (get_dist(current_turf, end) <= mintargetdist)
+		if(current_turf == end || closeenough)
+			var/datum/jps_node/final_node = new(current_turf,unwind_node, steps_taken)
+			sources[current_turf] = original_turf
+			unwind_path(final_node)
+			return
+		else if(sources[current_turf]) // already visited, essentially in the closed list
+			return
+		else
+			sources[current_turf] = original_turf
+
+		if(unwind_node.number_tiles + steps_taken > maxnodedepth)
+			return
+
+		var/interesting = FALSE // have we found a forced neighbor that would make us add this turf to the open list?
+
+		switch(heading)
+			if(NORTH)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, WEST, NORTHWEST) || STEP_NOT_HERE_BUT_THERE(current_turf, EAST, NORTHEAST))
+					interesting = TRUE
+			if(SOUTH)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, WEST, SOUTHWEST) || STEP_NOT_HERE_BUT_THERE(current_turf, EAST, SOUTHEAST))
+					interesting = TRUE
+			if(EAST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, NORTH, NORTHEAST) || STEP_NOT_HERE_BUT_THERE(current_turf, SOUTH, SOUTHEAST))
+					interesting = TRUE
+			if(WEST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, NORTH, NORTHWEST) || STEP_NOT_HERE_BUT_THERE(current_turf, SOUTH, SOUTHWEST))
+					interesting = TRUE
+
+		if(interesting)
+			var/datum/jps_node/newnode = new(current_turf, unwind_node, steps_taken)
+			openc[current_turf] = newnode
+			open.Insert(newnode)
+			return
```

Comment:
```suggestion
		if(!interesting)
			return
		var/datum/jps_node/newnode = new(current_turf, unwind_node, steps_taken)
		openc[current_turf] = newnode
		open.Insert(newnode)
```

Reply:
Your suggestion changes the code's behavior, this is in the context of a loop that would otherwise continue looping